### PR TITLE
payment history page back button to previous page

### LINKED ISSUE #96174: [Payment History] Back Button Not Handling Traversing Backwards in Pagination Tabs
# Problem

On the Payment History page, when the user utilizes pagination to advance through the Payment History table, pressing the Back button on the browser does not take them to the page in the table they were previously on.  Instead, it navigates them backwards to the link they followed to come to the Payment History page.

# Solution

Pressing the back button when pagination is being used should return the user to the previous page in the table they were on, not navigate to the previous browser page.

# Steps to Reproduce

1. Log into https://staging.va.gov with a user who has multiple pages of payment history data (e.g. vets.gov.user+229@gmail.com)
2. Navigate to the [Payment History page](https://staging.va.gov/va-payment-history/payments/)
3. Select page 2 or greater on the Payment History table
4. Press the back button

**Expected**: User returns to Page 1 of the Payment History table
**Actual**: The user is taken back to the previous page of the website they were viewing

# Notes

BMT 1 just completed something similar for another page; possibly reach out to them for a link to that issue and see how the problem was solved in that instance.

# Screenshots/Recordings
https://github.com/user-attachments/assets/42adccd5-cc0d-47fe-a09f-8578343ba56d

### PR DESCRIPTION
**Pull request by janechodance**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Add navigation to payment history page so the back button will correctly navigate to previous page

## Related issue(s)
[department-of-veterans-affairs/va.gov-team#96174](https://github.com/department-of-veterans-affairs/va.gov-team/issues/96174)

## Testing done

locally tested and add unit test

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

https://github.com/user-attachments/assets/17aed935-14eb-4e5a-95d6-ce11001aedc0

## What areas of the site does it impact?

Payment history page

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed. [#980689](https://github.com/department-of-veterans-affairs/va.gov-team/issues/98069)

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by dysbo — APPROVED** [review]
I like managing the page solely through the param vs managing it in both the param and the state. Great move.

**Review by dysbo — DISMISSED** [review]

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Comment by 1Copenut** [comment]
Starting accessibility test issue now. Should have findings in an hour or two.

File: src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx [hunk 1]
```diff
@@ -26,22 +27,24 @@ const Payments = ({
...
+      setCurrentData(paginatedData[currentPage - 1]);
+      setTotalPages(paginatedData.length);
+    },
+    [currentPage, data],
+  );
 
   const onPageChange = page => {
-    setCurrentData(paginatedData.current[page - 1]);
-    setCurrentPage(page);
+    const newURL = `${location.pathname}?page=${page}`;
```
Comment (1Copenut) [inline src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx:46]: I'd like to use this custom event to also set focus on the H3 that announces the number of results. I tested it locally and it's working really well and only a few more lines:

```diff
// Line 33
+ const tableHeadingRef = useRef(null);

const onPageChange = page => {
  const newURL = `${location.pathname}?page=${page}`;
+ if (tableHeadingRef) {
+   tableHeadingRef.current.focus();
+ }
  navigate(newURL);
};

<h3
  className="vads-u-font-size--lg vads-u-font-family--serif"
+ ref={tableHeadingRef}
+ tabIndex={-1}
>
```

**Review by 1Copenut — COMMENTED** [review]
Hi @janechodance. I'm still working with DST to figure out this phantom axe-core issue, but otherwise the accessibility review has gone really well. I added an improvement (suggested) change for screen reader handling.

If you'd prefer to add this as a separate issue, let me know and I'll file a follow-on ticket.

Thank you!

**Review by 1Copenut — DISMISSED** [review]
👍LGTM! I retested with latest `main` and the axe browser plugin issue we were seeing before is gone. I retested the new focus management with NVDA and Narrator. Works great!

**Review by dysbo — DISMISSED** [review]

File: src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx [hunk 2]
```diff
@@ -26,22 +27,28 @@ const Payments = ({
...
+  const location = useLocation();
+  const navigate = useNavigate();
   const [currentData, setCurrentData] = useState([]);
-  const [currentPage, setCurrentPage] = useState(1);
-  // Using `useRef` here to avoid triggering a rerender whenever these are
-  // updated
-  const totalPages = useRef(0);
-  const paginatedData = useRef([]);
+  const currentPage = new URLSearchParams(location.search).get('page') || 1;
+  const [totalPages, setTotalPages] = useState(0);
```
Comment (jerekshoe) [inline src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx:34]: I would use `useRef` here unless you want the page to re-render when you call `setTotalPages`

**Review by dysbo — DISMISSED** [review]

**Review by dysbo — APPROVED** [review]

**Review by IGallupSoCo — APPROVED** [review]
LGTM - you're good to move forward on the merge

**Pull request merged**